cpp/src/parquet/file_writer.cc

```diff
@@ -359,14 +359,26 @@ class FileSerializer : public ParquetFileWriter::Contents {
     if (row_group_writer_) {
       row_group_writer_->Close();
     }
+    int16_t row_group_ordinal = 0;
+    if (num_row_groups_ < std::numeric_limits<int16_t>::max()) {
+      row_group_ordinal = static_cast<int16_t>(num_row_groups_);
```

Comment:
This check is a bit ugly. Maybe we can also not write the row_group_ordinal for un-encryption files

Reply:
Yes, I think there is no need to write it.